scripts/CSSRE/lib/py/kafka.py

```diff
@@ -81,23 +81,24 @@ def run_kafka_topics_script(self, namespace, kafka, topic="", filter="") :
     Parameters:
       namespace (str): Namespace of the kafka.
       kafka (str): Name of the kafka cluster.
+      service (str): Name of the service being used to run oc exec against StatefulSet or StrimziPodSet.
       filter (str): a filter for partitions (--under-replicated-partitions, --under-min-isr-partitions, --unavailable-partitions).
 
     Returns
       (apiobject): OpenShift API object with description of topics and partitions.
     """
     with self._oc.project(namespace), self._oc.timeout(self.settings.OC_TIMEOUT_5_MINUTES):
 
-      cmd = "-i statefulset/"+kafka+"-kafka -c kafka -- env - bin/kafka-topics.sh --bootstrap-server localhost:9096 --describe "
+      cmd = "-i {service} -c kafka -- env - bin/kafka-topics.sh --bootstrap-server localhost:9096 --describe "
       if topic:
         cmd = cmd + " --topic " + topic
       if filter:
         cmd = cmd + " --" + filter
       try:
         return self._oc.invoke('exec', cmd.split())
       except self._oc.OpenShiftPythonException:
-        raise StatefulsetExecConnectionError(
-          f"Connection to Statefulset failed: statefulset/{kafka}-kafka"
+        raise ServiceExecConnectionError(
+          f"Connection to Service failed: {service}" #service name
```

Comment:
Is this giving enough information?

Reply:
Extra information is provided by the custom exception defined in https://github.com/openshift/managed-scripts/blob/main/scripts/CSSRE/lib/py/exceptions.py

Here you're replacing the `StatefulsetExecConnectionError` with a new one, `ServiceExecConnectionError`. You'll need to add the new custom exception to the above file (or just modify `StatefulsetExecConnectionError` if it's not being used anymore). You can provide extra information about what this exception means there.

Reply:
@robshelly Hi Rob, thank you. I just pushed some new changes and I think I've changed it to what you're saying here, would you mind taking a look when you get the chance? :) Thanks!